### PR TITLE
libpst: build as shared library, evolution: enable importing Outlook .pst data files

### PR DESCRIPTION
**Pull request by mvnetbiz**
###### Motivation for this change
I wanted to import outlook email .pst files into evolution mail.

###### Things done
Build libpst shared, add it to evolution buildInputs
```term
$ nix path-info -S evolution-old/
/nix/store/8p3ii0dwbsrx6dxrba1zzkh6mp1mi9gl-evolution-3.38.4	 1199130256
$ nix path-info -S evolution-new/
/nix/store/f11sln8v1zgc5z79f5mbj4wg3ksjxfp4-evolution-3.38.4	 1202443944
```
```term
$ nix path-info -S libpst-old/
/nix/store/1krnd7jmj84cyzcvb3wkncss1hiwn1j8-libpst-0.6.75	   67126848
$ nix path-info -S libpst-new/
/nix/store/ynvizdy0r3f3b1q1j33mmcfkk2l0vv51-libpst-0.6.75	   66852816
```

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117424` at 29f982cb run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>almanah</li>
    <li>evolution</li>
    <li>evolution-ews</li>
    <li>libpst</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

doxygen is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/libraries/libpst/default.nix:27:3:
```
   |
27 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

xmlto is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/libraries/libpst/default.nix:27:3:
```
   |
27 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/mailreaders/evolution/evolution/default.nix:118:5:
```
    |
118 |     ./moduledir_from_env.patch
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/libpst/default.nix:47:5:
```
   |
47 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/libraries/libpst/default.nix [hunk 1]
```diff
@@ -9,14 +19,24 @@ stdenv.mkDerivation rec {
...
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+  ];
+
   buildInputs = [
-    boost libgsf bzip2
-    xmlto gettext imagemagick doxygen
+    bzip2
+    doxygen
```
Comment (jtojnar) [inline pkgs/development/libraries/libpst/default.nix:29]: `doxygen` and `xmlto` are likely used for building docs so they should go to `nativeBuildInputs`. And I would expect the same for gettext. And imagick, but that one is actually used at runtime: http://hg.five-ten-sg.com/libpst/file/tip/src/pst2dii.cpp.in#l36

File: pkgs/development/libraries/libpst/default.nix [hunk 2]
```diff
@@ -25,7 +45,7 @@ stdenv.mkDerivation rec {
     homepage = "https://www.five-ten-sg.com/libpst/";
     description = "A library to read PST (MS Outlook Personal Folders) files";
     license = licenses.gpl2;
```
Comment (jtojnar) [inline pkgs/development/libraries/libpst/default.nix:47]: Looks like `gpl2Plus`: http://hg.five-ten-sg.com/libpst/file/tip/xml/libpst.in#l287

File: pkgs/development/libraries/libpst/default.nix (hunk 1 above)
Comment (mvnetbiz) [reply pkgs/development/libraries/libpst/default.nix:29]: For imagemagick do I put it in propagatedBuildInputs, or keep it in buildInputs? It looks like it sets the full path at build time, so I don't think I should remove it entirely right? I always get confused by dependencies, thanks for the help.

Comment (jtojnar) [reply pkgs/development/libraries/libpst/default.nix:29]: I think `imagemagick` can stay in `buildInputs` since it will be hardcoded.

**Review by jtojnar — APPROVED** [review]

**Comment by jtojnar** [comment]
Thanks.

**Pull request merged**